lib/trivial.nix

```diff
@@ -467,6 +467,17 @@ rec {
     then v
     else k: v;
 
+  /* Convert a hexadecimal string to it's integer representation. For example:
+
+    fromHexString "FF" => 255
+
+    fromHexString (builtins.hashString "sha256" "test") => 9223372036854775807
+  */
+  fromHexString = value:
+  let
+    parsed = builtins.fromTOML "v=0x${value}";
```

Comment:
We should make sure to check for the limits of this approach. I recommend a single `builtins.match "^[0-9a-fA-F]{,<maximum>}$" value` to validate there's no invalid characters and not too many (fill in <maximum>). The current length limit should also be documented as a warning in the docs (see [here](https://github.com/NixOS/nixpkgs/tree/master/doc#admonitions)), with a note that it could get increased in the future.